ip/opus.c

```diff
@@ -232,14 +232,25 @@ static int opus_read_comments(struct input_plugin_data *ip_data,
 	GROWING_KEYVALS(c);
 	struct opus_private *priv;
 	const OpusTags *ot;
+	const OpusHead *head;
 	int i;
 
 	priv = ip_data->private;
 
+	head = op_head(priv->of, -1);
+	if(head != NULL) {
+		char *val = xmalloc0(12); // 11 max int digits + NULL
```

Comment:
Since the `output_gain` field of the `OpusHead` struct is an int, I thought I'd allocate the memory for all possible ints, but I guess a range check would be better?

Reply:
Or just decreasing the allocated memory if the range is guaranteed (which I'm not sure if it is)

Reply:
For reference I was quoting https://opus-codec.org/docs/opusfile_api-0.5/structOpusHead.html . I wouldn't add a check, 12 isn't a problem really.

Reply:
Yeah, I didn't think 12 bytes wasn't too much, so used that just in case.